### DISCUSSION TITLE
限定共有記事を公開しようとするとエラーになる

### DISCUSSION THREAD

**Discussion by s2terminal** [increments/qiita-discussions#293] [要望・アイデア・不具合]
不具合の可能性がある挙動を見つけましたので、共有致します。

## 概要
限定共有状態になっている記事を公開にしようとすると、添付画像のようなエラーが表示されます。

![image](https://user-images.githubusercontent.com/7953751/189060966-04184088-d9c4-4818-b402-9d3f76fd780f.png)

## 再現手順
1. 限定共有状態の記事を開き、「編集する」
2. 編集画面で「内容を確定」
3. 「変更内容の概要(任意)」の画面で「更新する」

期待する結果→記事が更新される

実際の挙動→添付の「Secretcan't be secret with organization」というエラーが画面左下に表示され、保存できない

ご確認よろしくお願い致します。

**Comment by tanaka-asami415** [comment]
@s2terminal 
ご報告ありがとうございます🙏
こちら調査して対応いたします🙇

**Reply by tanaka-asami415** [reply]
@s2terminal
こちら修正が完了しましたので、公開ができるかお試しいただけますでしょうか🙇 
[下書き一覧](https://qiita.com/drafts) に編集途中の内容が残っていると思いますので、
そちらから編集していただき、お試しいただけますと幸いです🙇

**Reply by s2terminal** [reply]
対応ありがとうございます。
試したところ同様のエラーが再度か発生したのですが、何度か試していたら成功しました。

再現手順が分からず恐縮なのですが、成功した時はまず記事にOrganizationが紐付けられていない状態で一度公開し、再度編集してOrganizationを紐付ける、という手順を取りました。
Organizationの選択は記事の編集後の保存時にセレクトボックスで選択できると思うのですが、このセレクトボックスが出る場合と出ない場合があり、条件は分かりませんがエラーに関係していそうな印象を受けました。

**Reply by tanaka-asami415** [reply]
@s2terminal 
ご確認いただきありがとうございます🙇
ご不便をおかけし申し訳ありません。引き続き調査いたします🙇

**Reply by s2terminal** [reply]
別の記事で検証し、以下の手順にてエラーが再現することを確認しました。

1. Qiitaで記事を書き、「限定共有投稿」で投稿する
2. 該当の記事の編集画面を開き、右下の緑色の「内容を確定」ボタンの右側にある「▲」を押し、出てくる「→内容を確定」を押す
    - ![image](https://user-images.githubusercontent.com/7953751/189323057-ec08cd51-fc80-4894-8d96-7d98ac28e0a8.png)
3. 右下の緑色の「内容を確定」ボタンを押す
4. 限定共有投稿に対して通常は表示されないと思われる「Organizationの紐付け」がセレクトボックス表示される
    - ![image](https://user-images.githubusercontent.com/7953751/189323447-5950a554-98ce-4eed-8ba6-c0aa7778deca.png)
5. Organizationを紐付けて「更新する」を押すと「Secretcan't be secret with organization」エラーが発生する

また、本事象が発生した記事を、編集せず閲覧画面から「全体に公開する」ことでエラーが解消されることも確認しました。

**Reply by wataru86** [reply]
@s2terminal
限定共有記事の場合にOrganization選択のセレクトボックスが出ないように修正しました！
詳細な再現手順のおかげで修正できました！ありがとうございます！

**Reply by s2terminal** [reply]
対応ありがとうございました！
